### PR TITLE
Add notes from the 1.2 release party (the real 1.2 release)

### PR DESCRIPTION
**Pull request by maelvls**
Signed-off-by: Maël Valais <mael@vls.dev>

### CODE REVIEW

File: content/en/docs/contributing/release-process.md [hunk 1]
```diff
@@ -62,128 +72,163 @@ The process for cutting a minor release is as follows:
...
 
 3. Push it to the `jetstack/cert-manager` repository
 
-   ```bash
-   git push --set-upstream origin
-   ```
+     Note 1: run `git remote -v` to check that `origin` points to the
+     upstream <https://github.com/jetstack/cert-manager.git>.
+
+     Note 2: uou need to be an "admin" of the GitHub project to be able to
```
Comment (jakexks) [inline content/en/docs/contributing/release-process.md:99]:
```suggestion
     Note 2: you need to be an "admin" of the GitHub project to be able to
```

File: content/en/docs/contributing/release-process.md [hunk 2]
```diff
@@ -15,28 +15,38 @@ perform a cert-manager release:
 
 1. Install the [`release-notes`](https://github.com/kubernetes/release/blob/master/cmd/release-notes/README.md) CLI:
    ```sh
-   (cd && GO111MODULE=on go get k8s.io/release/cmd/release-notes@latest)
+   (cd && GO111MODULE=on go get k8s.io/release/cmd/release-notes@v0.7.0)
    ```
 2. Install our [`cmrel`](https://github.com/cert-manager/release) CLI:
```
Comment (jakexks) [inline content/en/docs/contributing/release-process.md:20]: This should be clone the cmrel repo as everything needs to be run from within the checked-out repository

Comment (maelvls) [reply content/en/docs/contributing/release-process.md:20]: I added a new line with the warning "remember to stay in the cloned `release/` folder while typing all the commands",m what do you think?

**Pull request merged**